packages/medusa/src/services/customer.ts

```diff
@@ -195,9 +200,28 @@ class CustomerService extends TransactionBaseService {
     email: string,
     config: FindConfig<Customer> = {}
   ): Promise<Customer | never> {
-    return await this.retrieve_({ email: email.toLowerCase() }, config)
+    return await this.retrieve_(
+      { email: email.toLowerCase(), has_account: false },
```

Comment:
comment: This is a potentially breaking change no? We might have developers relying on `retrieveByEmail` to return a registered customer.

If so, we should add `BREAKING CHANGE: [description of change]` in the changesets.

Reply:
Alternatively, we could keep this one as deprecated and add `retrieveUnregisteredByEmail`

Reply:
+ bump this to be a minor in the changeset.

Reply:
marking it deprecated